test/testinotifywatcher.cpp

```diff
@@ -11,18 +11,23 @@
 
 using namespace OCC;
 
-class TestInotifyWatcher: public FolderWatcherPrivate
+
+struct FriendlyFolderWatcherPrivate : FolderWatcherPrivate
 {
-    Q_OBJECT
+    using FolderWatcherPrivate::FolderWatcherPrivate;
+    friend class TestInotifyWatcher;
+};
 
-private:
-    QString _root;
 
+class TestInotifyWatcher : public QObject
+{
+    Q_OBJECT
 private slots:
-    void initTestCase() {
-        qsrand(QTime::currentTime().msec());
+    // Test the recursive path listing function lists everything
+    void testAddFolderRecursiveHelper() {
+        QTemporaryDir tmpDir;
 
-        _root = QDir::tempPath() + "/" + "test_" + QString::number(qrand());
```

Comment:
I'm using QTemporaryDir which take care of that instead of manually creating and destroying the directory.